### ISSUE TITLE
Source document at runtime should be removed

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#423]
Command line and Ant provide a source document when running the tests. But it can be harmful. For example,

**do-nothing.xsl**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<xsl:stylesheet version="2.0" xmlns:xsl="http://www.w3.org/1999/XSL/Transform" />
```

**test.xspec**
```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:description stylesheet="do-nothing.xsl" xmlns:x="http://www.jenitennison.com/xslt/xspec">
	<x:scenario>
		<x:label>Suppose you're testing count() function with child::element() selected but you
			forget to provide the selection context (either inner XML or @href)</x:label>
		<x:call function="count">
			<x:param select="child::element()" />
		</x:call>
		<x:expect label="and you happen to expect 1" select="1" />
	</x:scenario>
</x:description>
```

This `x:param` doesn't make sense, but `x:expect` returns Success without any complaint, because `x:param/@select` is evaluated in the source document node of the .xspec file.
So the source document can be harmful.

The purpose of the source document is discussed in expath/xspec#73. No use case is presented there.
On the other hand, I have one use case:
```xml
	<x:call function="my:detect-text-encoding">
		<x:param select="
			'resources/UTF-8.bin'
			=> resolve-uri(document-uri())
			=> saxon:read-binary-resource()" />
```
This way you can read the binary files relative to the .xspec file. I sometimes use this trick in my production XSpec files.

To sum up, I think
* The source document at runtime can be harmful or confusing at best. Command line and Ant should remove it.
* XSpec should provide an alternative way to get the URI of the .xspec file at runtime.
